src/Microsoft.AspNetCore.Sockets.Client.Http/ITransport.cs

```diff
@@ -9,7 +9,7 @@ namespace Microsoft.AspNetCore.Sockets.Client
 {
     public interface ITransport
     {
-        Task StartAsync(Uri url, Channel<byte[], SendMessage> application, TransferMode requestedTransferMode, string connectionId);
+        Task StartAsync(Uri url, Channel<byte[], SendMessage> application, TransferMode requestedTransferMode, string connectionId, IConnection connection);
```

Comment:
This isn't very clean. We shouldn't need to pass the IConnection here.

Reply:
How else should transports add features? I actually think it makes sense, it's much like how it works on the server.